### PR TITLE
Template text editor | From RTE

### PR DESCRIPTION
**Pull request by Yossitrx**
## Status 
**READY**

## Description
Converts the content From RTE Server compatible. 

## Todos
- [x] Tests

## Impacted Areas in Application
BOB-STYLE

### CODE REVIEW

File: projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts [hunk 1]
```diff
@@ -1,22 +1,41 @@
...
       });
   }
 
-  private matchListOptionsToConvertedValue(placeholders: Placeholder[], id: string): string {
+  private matchPlaceholderToConvertedValueBtId(placeholders: Placeholder[], id: string): string {
     const placeholder = find(placeholders, p => p.id === id);
     return placeholder ? placeholder.displayName : id;
   }
+
+  private matchPlaceholderToConvertedValueBtDisplyName(placeholders: Placeholder[], displayName: string): string {
```
Comment (tomersela) [inline projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts]: We don't need this, parse it from the HTML.

File: projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts [hunk 2]
```diff
@@ -1,22 +1,41 @@
...
 @Injectable()
 export class PlaceholderRteConverterService {
   constructor() {
   }
 
+  public fromRte(rteInnerHtml: string, placeholders: Placeholder[]): string {
+    const elm: HTMLElement = document.createElement('div');
+    elm.innerHTML = rteInnerHtml + '';
+    Array.from(elm.querySelectorAll('[placeholder]')).forEach(e => {
+      const b: Text =
```
Comment (tomersela) [inline projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts]: I think we talked about this `b` already. use a meaningful name

**Comment by Yossitrx** [comment]
@tomersela Updated

File: projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts [hunk 3]
```diff
@@ -1,21 +1,33 @@
...
     return contentToConvert
       .replace(regex, (field: string, innerContent: string) => {
         // tslint:disable-next-line:max-line-length
-        return`<span placeholder="${innerContent}">${this.matchListOptionsToConvertedValue(placeholders, innerContent)}</span>`;
+        return`<span placeholder="${innerContent}">${this.matchPlaceholderToConvertedValueBtId(placeholders, innerContent)}</span>`;
       });
   }
 
-  private matchListOptionsToConvertedValue(placeholders: Placeholder[], id: string): string {
+  private matchPlaceholderToConvertedValueBtId(placeholders: Placeholder[], id: string): string {
```
Comment (tomersela) [inline projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts]: getDisplayNameById

File: projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts [hunk 4]
```diff
@@ -1,21 +1,33 @@
...
 export class PlaceholderRteConverterService {
   constructor() {
   }
 
+  public fromRte(rteInnerHtml: string): string {
+    const elm: HTMLElement = document.createElement('div');
+    elm.innerHTML = rteInnerHtml + '';
+    Array.from(elm.querySelectorAll('[placeholder]')).forEach(existingElement => {
+      const placeholderTextElement: Text =
+        document.createTextNode('{{' + existingElement.getAttribute('placeholder') + '}}');
```
Comment (tomersela) [inline projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts]: extract `existingElement.getAttribute('placeholder')` to a const

**Comment by Yossitrx** [comment]
@tomersela Updated

File: projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts [hunk 5]
```diff
@@ -1,21 +1,34 @@
...
+
 @Injectable()
 export class PlaceholderRteConverterService {
   constructor() {
   }
 
+  public fromRte(rteInnerHtml: string): string {
+    const elm: HTMLElement = document.createElement('div');
+    elm.innerHTML = rteInnerHtml + '';
+    Array.from(elm.querySelectorAll('[placeholder]')).forEach(existingElement => {
```
Comment (galloween) [inline projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts]: I suggest you change the attribute to 'data-placeholder' for 2 reasons
- 'placeholder' is an existing native html attribute
- you can use: existingElement.dataset.placeholder to get value of 'data-placeholder' attribute

Comment (tomersela) [reply projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts]: and it contains the id of the placeholder, so maybe we can be more specific - `data-placeholder-id`

Comment (galloween) [reply projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts]: in any case, by htm5 specs you should prefix custom attributes with 'data-'

Comment (Yossitrx) [reply projects/ui-framework/src/lib/form-elements/template-text-editor/placeholder-rte-converter.service.ts]: Cool, will change it.

**Pull request merged**